std/algorithm/mutation.d

```diff
@@ -1940,6 +1944,100 @@ if (isBidirectionalRange!Range
             [ 1, 3, 3, 4, 5, 5, 6 ]);
 }
 
+@nogc unittest
+{
+    // @nogc test
+    int[10] arr = [0,1,2,3,4,5,6,7,8,9];
+    static bool pred(int i)
+    {
+        if (i < 5)
+            return true;
+        return false;
+    }
+    auto r = arr[].remove!(SwapStrategy.unstable)(0);
+    r = r.remove!(SwapStrategy.stable)(0);
+    r = r.remove!(pred, SwapStrategy.unstable);
+    r = r.remove!(pred, SwapStrategy.stable);
+}
+
+unittest
+{
+    import std.typecons : Tuple;
+    import std.meta : AliasSeq;
+    import std.range : iota;
+    import std.array : array;
+    import std.algorithm.comparison : min;
+    alias S = Tuple!(int[2]);
+    S[] soffsets;
+    foreach (start; 0..5)
+    foreach (end; min(start+1,5) .. 5)
+          soffsets ~= S([start,end]);
+    alias D = Tuple!(int[2],int[2]);
+    D[] doffsets;
+    foreach (start1; 0..10)
+    foreach (end1; min(start1+1,10) .. 10)
+    foreach (start2; end1 ..10)
+    foreach (end2; min(start2+1,10) .. 10)
+          doffsets ~= D([start1,end1],[start2,end2]);
+    alias T = Tuple!(int[2],int[2],int[2]);
+    T[] toffsets;
+    foreach (start1; 0..15)
+    foreach (end1; min(start1+1,15) .. 15)
+    foreach (start2; end1..15)
+    foreach (end2; min(start2+1,15) .. 15)
+    foreach (start3; end2..15)
+    foreach (end3; min(start3+1,15) .. 15)
+            toffsets ~= T([start1,end1],[start2,end2],[start3,end3]);
+    static bool verify(O...)(int[] r, int len, int removed, bool stable, O offsets)
+    {
+        if (r.length != len - removed)
+            return false;
+        if (stable)
+        {
+            long prev = long.min;
+            foreach (i;r)
+            {
+                if (i < prev)
+                    return false;
+                prev = i;
+            }
+        }
+        foreach (i; r)
+            foreach (offset; offsets)
+                if (i >= offset[0] && i < offset[1])
+                    return false;
```

Comment:
If you want to avoid the "red", uncovered line you could do sth. like:

```d
        import std.algorithm.searching : any, canFind;
        import std.range : only;
        return r.any!(e => !offsets.only.canFind!(o => o[0] <= e && e < o[1]));
```